File: day1.rb
input = File.read('/Users/avylov/dev/adv_code_23/day1_part1_input.in').strip

part1 = 0
part2 = 0

input.each_line do |line|
  part1_digits = []
  part2_digits = []

  line.chars.each_with_index do |c, i|
    if c.match?(/\d/)
      part1_digits.push(c)
      part2_digits.push(c)
    end

    values = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    values.each_with_index do |val, d|
      if line[i..-1].start_with?(val)
        part2_digits.push((d + 1).to_s)
      end
    end
  end

  part1 += (part1_digits[0] + part1_digits[-1]).to_i
  part2 += (part2_digits[0] + part2_digits[-1]).to_i
end

puts part1
puts part2

File: day1.js
const fs = require('fs');

const input = fs.readFileSync('/Users/avylov/dev/adv_code_23/day1_part1_input.in', 'utf8').trimEnd();

let part1 = 0;
let part2 = 0;

input.split('\n').map(line => {
  const part1_digits = [];
  const part2_digits = [];
  
  for (let i = 0; i < line.length; i++) {
    const c = line[i];
    
    if (!isNaN(c)) {
      part1_digits.push(c);
      part2_digits.push(c);
    }
    
    const values = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
    for (let d = 0; d < values.length; d++) {
      if (line.substring(i).startsWith(values[d])) {
        part2_digits.push((d + 1).toString());
      }
    }
  }
  
  part1 += parseInt(part1_digits[0] + part1_digits[part1_digits.length - 1]);
  part2 += parseInt(part2_digits[0] + part2_digits[part2_digits.length - 1]);
});

console.log(part1);
console.log(part2);